### PR TITLE
MOSIP-44919 Update the release-0.18.x branch with latest automation code from develop

### PR DESCRIPTION
**Pull request by hegdenitin**
1. updating the release-0.18.x branch with latest automation code from develop branch for the release

## Release Notes - v0.18.0

* **New Features**
  * Added VP session request functionality with new verification endpoints
  * Introduced VC verification capability for DIDs with multiple signature algorithms
  * Enhanced request parameter handling to support transaction identifiers and validation flags
  * Added support for verifiable credential verification with status checks and claim filtering

* **Tests**
  * Expanded test coverage with comprehensive test cases for new VP session workflows
  * Added extensive VC verification test scenarios including negative cases and edge conditions
  * Introduced test cases for credential verification across multiple DID methods

* **Chores**
  * Updated version to 0.18.0-SNAPSHOT
  * Updated dependent library to released version

### CODE REVIEW

File: api-test/README.md [hunk 1]
```diff
@@ -98,7 +98,7 @@ To execute the tests using Jar, use the following steps:
 
 2. Run the automation test suite JAR file:
    ```
-   java -Dmodules=injiverify -Denv.user=api-internal.<env_name> -Denv.endpoint=<base_env> -Denv.testLevel=smokeAndRegression -jar apitest-injiverify-0.17.0-jar-with-dependencies.jar
+   java -Dmodules=injiverify -Denv.user=api-internal.<env_name> -Denv.endpoint=<base_env> -Denv.testLevel=smokeAndRegression -jar apitest-injiverify-0.18.0-jar-with-dependencies.jar
```
[the remark this answers is not in the document]
Comment (hegdenitin) [reply api-test/README.md:101]: tested and automation is working fine - ignorable

File: api-test/src/main/java/io/inji/testrig/apirig/injiverify/testscripts/SimplePostForAutoGenId.java [hunk 2]
```diff
@@ -88,22 +97,69 @@ public void test(TestCaseDTO testCaseDTO)
 		}
 
 		String inputJson = getJsonFromTemplate(testCaseDTO.getInput(), testCaseDTO.getInputTemplate());
+
 		inputJson = inputJsonModuleKeyWordHandler(inputJson, testCaseName);
+		inputJson = InjiVerifyUtil.replaceVpSessionInputPlaceholders(inputJson);
```
[the remark this answers is not in the document]
Comment (hegdenitin) [reply api-test/src/main/java/io/inji/testrig/apirig/injiverify/testscripts/SimplePostForAutoGenId.java:102]: tested and automation is working fine - ignorable

File: api-test/src/main/java/io/inji/testrig/apirig/injiverify/testscripts/SimplePostForAutoGenId.java [hunk 3]
```diff
@@ -88,22 +97,69 @@ public void test(TestCaseDTO testCaseDTO)
...
+		if (inputJson != null) {
+			try {
+				JSONObject requestJson = new JSONObject(inputJson);
+
+				if (requestJson.has(JSON_KEY_COOKIE)) {
+					cookieValue = requestJson.optString(JSON_KEY_COOKIE);
+
+					if (cookieValue != null && !cookieValue.isEmpty()) {
+						logger.info("Sending cookie via HEADER: " + cookieValue);
+
```
[the remark this answers is not in the document]
Comment (hegdenitin) [reply api-test/src/main/java/io/inji/testrig/apirig/injiverify/testscripts/SimplePostForAutoGenId.java:115]: tested and automation is working fine - ignorable

File: api-test/src/main/resources/config/testCaseInterDependency.json [hunk 4]
```diff
@@ -14,5 +14,7 @@
     "TC_InjiVerify_GetVpRequestStatus_01": ["TC_InjiVerify_CreateNewVerificationRequest_01"],
     "TC_InjiVerify_VpSubmission_07": ["TC_InjiVerify_CreateNewVerificationRequest_09"],
     "TC_InjiVerify_VpSubmission_11": ["TC_InjiVerify_CreateNewVerificationRequest_01"],
-    "TC_InjiVerify_VpSubmission_10": ["TC_InjiVerify_CreateNewVerificationRequest_09"]
-}
\ No newline at end of file
+    "TC_InjiVerify_VpSubmission_10": ["TC_InjiVerify_CreateNewVerificationRequest_09"],
+    "TC_InjiVerify_VpSubmissionECR1_01": ["TC_InjiVerify_CreateNewVerificationRequest_10"],
```
[the remark this answers is not in the document]
Comment (hegdenitin) [reply api-test/src/main/resources/config/testCaseInterDependency.json:18]: tested and automation is working fine - ignorable

File: api-test/src/main/resources/injiverify/CreateNewVerificationRequest/CreateNewVerificationRequestECCR1.hbs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        "constraints": {
+          "fields": [
+            {
+              "path": [
+                "$.type"
+              ],
+              "filter": {
+                "type": "object",
+                "pattern": "InsuranceCredential"
+              }
```
[the remark this answers is not in the document]
Comment (hegdenitin) [reply api-test/src/main/resources/injiverify/CreateNewVerificationRequest/CreateNewVerificationRequestECCR1.hbs:29]: tested and automation is working fine - ignorable

File: api-test/src/main/resources/injiverify/VcVerificationDid/VcVerificationDid.yml [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+      outputTemplate: injiverify/VcVerificationDid/VcVerificationDidResultV2
+      input: '{
+        "expirationDate": "2020-11-17T06:06:36.456Z",
+        "skipStatusChecks": true,
+        "statusCheckFilters": [],
+        "includeClaims": true
+      }'
+      output: '{
+      "allChecksSuccessful": false
+}'
```
[the remark this answers is not in the document]
Comment (hegdenitin) [reply api-test/src/main/resources/injiverify/VcVerificationDid/VcVerificationDid.yml:125]: tested and automation is working fine - ignorable

File: api-test/src/main/resources/injiverify/VpRequestWithPresentationId/VpRequestWithPresentationId.yml [hunk 7]
```diff
@@ -5,32 +5,88 @@ VpRequestWithPresentationId:
...
+    input: '{
+      "clientId": "injiverify.mosip.net",
+      "nonce": "MTc2Mzk1NTY5NDI5NQ==",
+      "acceptVPWithoutHolderProof": true,
+      "responseCodeValidationRequired": "$IGNORE$",
+      "id": "$PRESENTATIONDEFINITIONID$"
+    }'
+    output: '{
+      "status": "400"
+}'
```
[the remark this answers is not in the document]
Comment (hegdenitin) [reply api-test/src/main/resources/injiverify/VpRequestWithPresentationId/VpRequestWithPresentationId.yml:92]: tested and automation is working fine - ignorable

File: api-test/src/main/resources/injiverify/VpRequestWithPresentationId/VpRequestWithPresentationIdForVpSessionRequest.hbs [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        "constraints": {
+          "fields": [
+            {
+              "path": [
+                "$.type"
+              ],
+              "filter": {
+                "type": "object",
+                "pattern": "LifeInsuranceCredential"
+              }
```
[the remark this answers is not in the document]
Comment (hegdenitin) [reply api-test/src/main/resources/injiverify/VpRequestWithPresentationId/VpRequestWithPresentationIdForVpSessionRequest.hbs:28]: tested and automation is working fine - ignorable

**Review by mayuradesh — APPROVED** [review]
